pkg/api/types.go

```diff
@@ -653,7 +653,6 @@ type ServiceSpec struct {
 	PublicIPs []string `json:"publicIPs,omitempty"`
 
 	// ContainerPort is the name of the port on the container to direct traffic to.
-	// Optional, if unspecified use the first port on the container.
```

Comment:
Restored.  This has bad validation behavior if you don't put explicit ports on a Pod, I'll beef up the validation logic in a separate PR.  Filed: https://github.com/GoogleCloudPlatform/kubernetes/issues/3838